src/__Private/Inspector/InspectorCLI.hack

```diff
@@ -76,12 +76,42 @@ final class InspectorCLI extends CLIWithRequiredArguments {
     print $output."\n";
 
     if ($this->open) {
-      \pcntl_exec('/usr/bin/open', [$output]);
+      $os_found = self::openFileInBrowser($output);
+      if (!$os_found) {
+        await $err->writeAsync('We could not determine your OS.');
+      }
     }
 
     return 0;
   }
 
+  private static function openFileInBrowser(string $filename): bool {
+    $os = get_host_os();
+    if ($os is null) {
+      return false;
+    }
+
+    /*HH_FIXME[2050] Use of a super global*/
+    $env = $_ENV;
+
+    switch ($os) {
+      case OperatingSystem::MACOS:
+        \pcntl_exec('/usr/bin/open', [$filename], $env);
+        break;
+      case OperatingSystem::LINUX:
+        using ($no_warnings = new PHPWarningSuppressor()) {
```

Comment:
should be able to just do `using new PHPWarningSuppressor`

If the assignment is required or you want code-as-documentation, please rename to `$_no_warnings`

Reply:
I only want to suppress this warning, since we have a good fallback. The others should still print the php warnings to stderr to give information to the user.